worker/docker_client.py

```diff
@@ -123,8 +134,18 @@ def _init_libcuda(self):
         # Find all the NVIDIA device files.
         self._nvidia_device_files = []
         for filename in os.listdir('/dev'):
-            if filename.startswith('nvidia'):
+            m = re.match(r'nvidia(.*)', filename)
+            if m is not None and (self._cuda_visible_devices is None or
```

Comment:
For clarity, define `device = m.group(1)`, and maybe match `(\d+)` instead of `(.*)`